client/src/utils/auth.js

```diff
@@ -31,15 +72,101 @@ export const saveUserIDToken = async (value) => {
 }
 
 /**
- * Removes a user's Google ID Token from Async Storage
+ * Saves a user's Google Refresh Token to SecureStore
+ * @param {String} value The users Google Refresh Token
+ * @returns {Boolean} true if the operation was successful
+ */
+export const saveUserRefreshToken = async (value) => {
+  try {
+    await SecureStore.setItemAsync(SECURE_STORAGE_REFRESH_TOKEN_KEY, value)
+    return true
+  } catch (e) {
+    console.error('saveUserRefreshToken(): ', e.message)
+    return false
+  }
+}
+
+/**
+ * Saves a user's Google Client Id to SecureStore
+ * @param {String} value The users Google Client Id
+ * @returns {Boolean} true if the operation was successful
+ */
+export const saveUserClientId = async (value) => {
+  try {
+    await SecureStore.setItemAsync(SECURE_STORAGE_CLIENT_ID_KEY, value)
+    return true
+  } catch (e) {
+    console.error('saveUserClientId(): ', e.message)
+    return false
+  }
+}
+
+/**
+ * Removes a user's Google ID Token from SecureStore
  * @returns {Boolean} true if the operation was successful
  */
 export const removeUserIDToken = async () => {
   try {
-    await AsyncStorage.removeItem(ASYNC_STORAGE_ID_TOKEN_KEY)
+    await SecureStore.deleteItemAsync(SECURE_STORAGE_ID_TOKEN_KEY)
     return true
   } catch (e) {
-    console.error('saveUserIDToken(): ', e.message)
+    console.error('removeUserIDToken(): ', e.message)
     return false
   }
 }
+
+/**
+ * Removes a user's Google Refresh Token from SecureStore
+ * @returns {Boolean} true if the operation was successful
+ */
+export const removeUserRefreshToken = async () => {
+  try {
+    await SecureStore.deleteItemAsync(SECURE_STORAGE_REFRESH_TOKEN_KEY)
+    return true
+  } catch (e) {
+    console.error('removeUserRefreshToken(): ', e.message)
+    return false
+  }
+}
+
+/**
+ * Removes a user's Google Client Id from SecureStore
+ * @returns {Boolean} true if the operation was successful
+ */
+export const removeUserClientId = async () => {
+  try {
+    await SecureStore.deleteItemAsync(SECURE_STORAGE_CLIENT_ID_KEY)
+    return true
+  } catch (e) {
+    console.error('removeUserClientId(): ', e.message)
+    return false
+  }
+}
+
+/**
+ * Refreshes the ID token for user using refreshToken from SecureStore
+ * @returns {String} new IdToken if refresh is successful
+ */
+export const refreshIDToken = async () => {
+  try {
+    const refreshToken = await loadUserRefreshToken()
+    const clientId = await loadUserClientId()
+    return axios
+      .post(GOOGLE_OAUTH_TOKEN_URL, {
+        grant_type: 'refresh_token',
+        refresh_token: refreshToken,
+        clientId,
+      })
+      .then(({ data: { id_token: idToken } }) => {
+        saveUserIDToken(idToken)
+        return Promise.resolve(idToken)
+      })
+      .catch((reason) => {
+        console.error('rejected reauth')
```

Comment:
would like to see a bit more descriptive console.error message indicating where the app failed and why